include/pybind11/pybind11.h

```diff
@@ -466,7 +466,15 @@ class cpp_function : public function {
             }
             if (rec->def) {
                 std::free(const_cast<char *>(rec->def->ml_doc));
```

Comment:
I left this as-is without touching this line, feel free to change if needed.

Reply:
Nah, python/cpython#22670 just got merged, and backported into python/cpython#22674, so we should be fine excluding 3.9.0 and having something specificially works with this exact patch version :-)